airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/query.py

```diff
@@ -0,0 +1,1387 @@
+            ],
+        )
+    ]
+
+    query_nodes: List[Field] = [
+        "__typename",
+        "id",
+        Field(name="currencyCode", alias="currency"),
+        Field(
+            name="transactions",
+            fields=[
+                "id",
+                "errorCode",
+                Field(name="parentTransaction", fields=[Field(name="id", alias="parentId")]),
+                "test",
+                "kind",
+                "amount",
+                Field(name="receiptJson", alias="receipt"),
+                "gateway",
+                Field(name="authorizationCode", alias="authorization"),
+                "createdAt",
+                "status",
+                "processedAt",
+                Field(name="totalUnsettledSet", fields=total_unsettled_set_fields),
+                "paymentId",
+                Field(name="paymentDetails", fields=payment_details),
+            ],
+        ),
+    ]
+
+    record_composition = {
+        "new_record": "Order",
+    }
+
+    def process_transaction(self, record: Mapping[str, Any]) -> Mapping[str, Any]:
+        # save the id before it's resolved
+        record["admin_graphql_api_id"] = record.get("id")
+        # unnest nested fields
+        parent_transaction = record.get("parentTransaction", {})
+        if parent_transaction:
+            record["parent_id"] = parent_transaction.get("parentId")
+        # str values to float
+        record["amount"] = float(record.get("amount"))
+        # convert dates from ISO-8601 to RFC-3339
+        record["processedAt"] = self.tools.from_iso8601_to_rfc3339(record, "processedAt")
+        record["createdAt"] = self.tools.from_iso8601_to_rfc3339(record, "createdAt")
+        # resolve ids
+        record["id"] = self.tools.resolve_str_id(record.get("id"))
+        record["parent_id"] = self.tools.resolve_str_id(record.get("parent_id"))
+        # remove leftovers
+        record.pop("parentTransaction", None)
+        # field names to snake case
+        total_unsettled_set = record.get("totalUnsettledSet", {})
+        if total_unsettled_set:
+            record["totalUnsettledSet"] = self.tools.fields_names_to_snake_case(total_unsettled_set)
+            # nested str values to float
+            record["totalUnsettledSet"]["presentment_money"]["amount"] = float(
+                total_unsettled_set.get("presentmentMoney", {}).get("amount")
+            )
+            record["totalUnsettledSet"]["shop_money"]["amount"] = float(total_unsettled_set.get("shopMoney", {}).get("amount"))
+        payment_details = record.get("paymentDetails", {})
+        if payment_details:
+            record["paymentDetails"] = self.tools.fields_names_to_snake_case(payment_details)
+        # field names to snake case for root level
+        record = self.tools.fields_names_to_snake_case(record)
+        return record
+
+    def record_process_components(self, record: Optional[Mapping[str, Any]] = None) -> Optional[Iterable[Mapping[str, Any]]]:
```

Comment:
can you clarify in what situations you'd expect a None record?

Reply:
Corrected.